src/main/java/org/traccar/protocol/TeltonikaProtocolDecoder.java

```diff
@@ -254,28 +246,16 @@ private static void register(int id, Set<String> models, BiConsumer<Position, By
         register(200, fmbXXX, (p, b) -> p.set("sleepMode", b.readUnsignedByte()));
         register(205, fmbXXX, (p, b) -> p.set("cid2g", b.readUnsignedShort()));
         register(206, fmbXXX, (p, b) -> p.set("lac", b.readUnsignedShort()));
-        register(232, fmbXXX, (p, b) -> p.set("cngStatus", b.readUnsignedByte() > 0));
-        register(233, fmbXXX, (p, b) -> p.set("cngUsed", b.readUnsignedInt() * 0.1));
-        register(234, fmbXXX, (p, b) -> p.set("cngLevel", b.readUnsignedShort()));
-        register(235, fmbXXX, (p, b) -> p.set("oilLevel", b.readUnsignedByte()));
+        register(232, fmbXXX, (p, b) -> p.set("cngStatus", b.readUnsignedByte() > 0)); //only available via additional CAN reader
+        register(233, fmbXXX, (p, b) -> p.set("cngUsed", b.readUnsignedInt() * 0.1)); //only available via additional CAN reader
+        register(234, fmbXXX, (p, b) -> p.set("cngLevel", b.readUnsignedShort())); //only available via additional CAN reader
+        register(235, fmbXXX, (p, b) -> p.set("oilLevel", b.readUnsignedByte())); //only available via additional CAN reader
         register(236, null, (p, b) -> {
             p.set(Position.KEY_ALARM, b.readUnsignedByte() > 0 ? Position.ALARM_GENERAL : null);
         });
         register(239, null, (p, b) -> p.set(Position.KEY_IGNITION, b.readUnsignedByte() > 0));
         register(240, null, (p, b) -> p.set(Position.KEY_MOTION, b.readUnsignedByte() > 0));
         register(241, null, (p, b) -> p.set(Position.KEY_OPERATOR, b.readUnsignedInt()));
-        register(246, fmbXXX, (p, b) -> {
-            p.set(Position.KEY_ALARM, b.readUnsignedByte() > 0 ? Position.ALARM_TOW : null);
-        });
-        register(247, fmbXXX, (p, b) -> {
-            p.set(Position.KEY_ALARM, b.readUnsignedByte() > 0 ? Position.ALARM_ACCIDENT : null);
-        });
-        register(249, fmbXXX, (p, b) -> {
-            p.set(Position.KEY_ALARM, b.readUnsignedByte() > 0 ? Position.ALARM_JAMMING : null);
-        });
-        register(252, fmbXXX, (p, b) -> {
-            p.set(Position.KEY_ALARM, b.readUnsignedByte() > 0 ? Position.ALARM_POWER_CUT : null);
-        });
```

Comment:
Can you please explain what the problem is with all these?

Reply:
Sure:
1. AVL 246 is towing alarm. It can have values of `0: No Towing` and `1: Towing`. Traccar only has a ALARM_TOW key so we cannot record the No Towing event:
![image](https://github.com/traccar/traccar/assets/59513748/f91d6468-ba5c-464b-bd20-bc6f23a4c47a)


2. AVL 247 is accident alarm. It can have values from 1 to 8. This code only records one `accident` value and we have no idea what the AVL actual value was.
![image](https://github.com/traccar/traccar/assets/59513748/9eca2ff7-87fc-4d7c-bcff-61e2c4262cd3)


3. AVL 249 is jamming alarm. It can have values of `0: Jamming Stop` and `1: Jamming Start`. Traccar only has a ALARM_JAMMING key so we cannot record the Jamming Stop event:
![image](https://github.com/traccar/traccar/assets/59513748/be9d29ef-1f6d-4e20-ade8-a43b086ed264)


4. AVL 252 is related to the device power. It can have values of `0: battery present` and `1: battery unplugged`:
![image](https://github.com/traccar/traccar/assets/59513748/474473cb-b762-4244-b570-8e2ac5c41c58)

Reply:
Traccar does not have "no towing" event or "jamming stopped" event. I think it's ok if we also record the original IO value, but I want supported events to be decoded properly.

Reply:
I am all for decoding properly the supported events but if we leave this code, we create the following issues:

1. If the device sends AVL 249 - jamming stopped, currently the event is entirely skipped. So we have no record of the event ever being sent, unless in the log file.
2. A workaround would be to record the original io as you suggest but then we will first unnecessarily duplicate data in the database, therefore in the in the device details too. Second, the jamming event will be decoded with the protocol and the no jamming has to be decoded manually which means inconsistent handling of the same AVL. 
The accepted best practice would be to either entirely decode the AVL or not at all and then use computed attributes to handle it in a consistent way.

Reply:
I disagree with your conclusion of the best practice. The best practice in this case would be to store both raw value and the alarm.

Reply:
Let's agree to disagree :)

How do you suggest to fix this because the currently important data is being entirely skipped and missing from the server.

Reply:
I already explained twice what I think it a good solution for this.

Reply:
Well, currently the original raw value is not stored so the code is bad.

Reply:
Yes I am. When I asked how do you suggest to fix this, I meant what are the suggested code change, not about the logic (keep both values).

Reply:
> Yes I am. When I asked about how do you suggest to fix this, I meant what are the suggested code changes, not about the logic (keep both values).

Ok, so we have device which plugs directly to obd and device which uses lv_can.. shame for teltonka that for same functions it will have different avl's don't think ever it will be used in both cases.. so you use one or another..

Reply:
For alarm's yes traccar only have event's only for actual case has, been detected.. but don't have status which means ending... But most use cases i think it's needed to detect event.. and it's not so worth to know how long it took.. so if it's jamming it will trigger alarm each time it's detects.. also with towing.. and crash.. all values list in teltonika means same, crash.. but with different scenarious..

Reply:
Well, actually LV-CAN is a separate addon device. It's like an enhancement and it can be used both with FMBXXX and non FMBXXX devices. 
So it makes sense that the standalone FMBXXX series send the throttle and so do the LV-CAN addon (with a different ID so that we know which is which).

Reply:
I am not sure I understand. 
All I am saying is that while the code in question captures some of the available alarms on Traccar, it so misses other important data like when the jamming/towing ended, all the device connected events and the crash data trace type. Between the two, the latter is worst because it cannot be fixed with computed attributes.
It took me two months to realize something is wrong. I am sure there are other users who are still unaware.
So the code needs correction one way or another.
Waiting for @tananaev to provide his code suggestion.

Reply:
Ok, so removing everything also is not solution.. also while we have those values decoded don't do any harm.

Reply:
No, there is a problem because currently Traccar is not decoding all data and it's virtually missing from the database. So it  needs to be fixed.

Reply:
Ok, could you give use case scenario where it could lead to problem?

Reply:
I've already explained in great details above.

Reply:
> I've already explained in great details above.

Yes but what's a point to get end state? Where is a problem without it? Usually it's important to get start, but not length of problem.. so i am asking that, in what cases it's must have to know event length.. 

And for crash.. does the type of crash is so important?  Truly I don't remember from which version it have been detailed.. before that it was only one case.. crash.  Does it mean if crash detected with sensor calibrated or not? How different is full crash from real and limited? :D also what does it means fake crash? 

You only telling facts about states, but don't bring any use case where it's so important.. 

What else i see that you are talking about database.. so it seams that you are working with some extra software which connects directly to database to get it's readings, so maybe you have problems from there? Because I don't see any problem's while using traccar stand alone..

So if you have real problem, could you share to me? Maybe i don't understand exactly what behavior should be acceptable

Reply:
Hi Anton, any update on fixing the data values that are currently being missed?

Reply:
I think there's a fundamental disagreement, so no updates really.

Reply:
What do you mean by fundamental disagreements?

Is it true that the server currently deletes the AVL 252 and doesn't record anything in the database if the value of the AVL is `0`, code below:

`register(252, fmbXXX, (p, b) -> {
            p.set(Position.KEY_ALARM, b.readUnsignedByte() > 0 ? Position.ALARM_POWER_CUT : null);
});`

Reply:
The disagreement is that you just want the raw data, but I think most people are interested in decoded attributes, even if it means they technically losing some information.